kasa/smartcam/modules/camera.py

```diff
@@ -64,6 +64,21 @@ def _get_credentials(self) -> Credentials | None:
 
         return None
 
+    def _get_stream_creds(
+        self, credentials: Credentials | None = None
+    ) -> tuple[str, str] | None:
+        if not self.is_on:
+            return None
```

Comment:
The URL should probably still be valid even if the device is off. I'm still able to get stream using my credentials after switching the camera off in HA.

Reply:
Removed